subsys/spm/Kconfig

```diff
@@ -61,6 +61,19 @@ config SPM_SERVICE_RNG
 	  This service allows it to request random numbers from the SPM.
 endif
 
+config SPM_ALLOW_NON_SECURE_RESET
+	bool "Allow Non-Secure reset"
+#FIXME: Change this to Default to `n` when the issue regarding doing a software
```

Comment:
This should not be default yes, for all nRF91 boards right? We should be able to leave it to "n" for the boards not affected.

Reply:
What do you mean by "not affected"?

Reply:
>Since there are boards which do not have a pin-reset such as
	  Thingy91 this option should be enabled by default. If not a debugger
	  would not be able to issue a reset from a non-secure region even if it
	  is configure to be secure.

So, why have this "default y" for all boards, including those that have a pin reset working?

Reply:
I get your point. @sigvartmh will probably correct me, but I think we need to update this comment since this issue will hinder programming through the normal developer tools for all nrf91 boards. In which case it should be `default y` for all nrf91 boards.

Note that I might have misunderstood something, so lets wait for @sigvartmh 's response. Anyway, either the docs or the default must change.

Reply:
AFAIK, the reboot issue itself affects every board (see Jira NRF91-313). The JLink debugger automatically falls back to a pin reset if the soft reset doesn't work, so programming works fine in practice if pin reset is routed.

We cannot make the assumption that any new/custom board has pin reset routed, so a `y` setting until NRF91-313 is fixed avoids possibly breaking programming by default on new boards.

Adding `default n` for the nRF91 DK, which does have pin reset, would make sense.

Reply:
@ioannisg: If you can figure a good scheme to allow it for all boards nRF91 boards without a pin-reset enabled sure, I could not. It's possible to do  `default y if nrf91xx && !nrf9160_pca10090` reasoning is that customers will create their own custom boards. We don't know if they have or have connected pin-reset as this information does not exists yet or anywhere. There is also no information in device-tree or Kconfig which gives this information-

So from supports perspective as long as the issue NRF91-313 exists it would be a lot of extra support cases with people not being able to program/reset their custom boards I assume. Maybe @Rallare could give some more input on this.

Reply:
Added `default n if BOARD_NRF9160_PCA10090` but other defaults would have to be added on a needed basis.

Reply:
reverted this from a security standpoint to be of for every board except Thingy as a fix will come out soon for nrfjprog.